### PR TITLE
Rename BuildConfiguration to BuildContext

### PR DESCRIPTION
**Pull request by chanseokoh**

### CODE REVIEW

File: jib-core/src/main/java/com/google/cloud/tools/jib/configuration/BuildContext.java [hunk 1]
```diff
@@ -241,12 +241,12 @@ public Builder setExecutorService(@Nullable ExecutorService executorService) {
     }
 
     /**
-     * Builds a new {@link BuildConfiguration} using the parameters passed into the builder.
+     * Builds a new {@link BuildContext} using the parameters passed into the builder.
      *
      * @return the corresponding build configuration
```
Comment (TadCordle) [inline jib-core/src/main/java/com/google/cloud/tools/jib/configuration/BuildContext.java]: the corresponding build context

File: jib-core/src/test/java/com/google/cloud/tools/jib/configuration/BuildContextTest.java [hunk 2]
```diff
@@ -234,7 +230,7 @@ public void testBuilder_missingValues() throws IOException {
 
     // All required fields missing
     try {
-      BuildConfiguration.builder().build();
+      BuildContext.builder().build();
       Assert.fail("Build configuration should not be built with missing values");
```
Comment (TadCordle) [inline jib-core/src/test/java/com/google/cloud/tools/jib/configuration/BuildContextTest.java]: Build context should not be built with missing values?

Comment (TadCordle) [reply jib-core/src/test/java/com/google/cloud/tools/jib/configuration/BuildContextTest.java]: Or maybe just change it to "BuildContext"

File: jib-core/src/main/java/com/google/cloud/tools/jib/configuration/BuildContext.java (hunk 1 above)
Comment (chanseokoh) [reply jib-core/src/main/java/com/google/cloud/tools/jib/configuration/BuildContext.java]: good catch

**Pull request merged**